### ISSUE TITLE
jira: Can't authenticate with kerberos

### ISSUE THREAD

**Issue by KKoukiou** [pycontribs/jira#424]
I am running the following code to establish Jira connection with kerberos authentication.

``` python
from jira import JIRA
import logging

JIRA_SERVER = 'https://myserver.example.com'

logging.basicConfig(level=logging.DEBUG)

jira = JIRA(
    options={
        'server': JIRA_SERVER,
        'verify': False
    },
    kerberos=True
)
```
I don't see some Error but when I try to get the user that I am connected with I get anonymous user

``` python
In [1]: jira.current_user()
/home/kkoukiou/sandbox/Jira/lib/python2.7/site-packages/urllib3/connectionpool.py:858: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
DEBUG:urllib3.connectionpool:https://projects.engineering.redhat.com:443 "GET /rest/api/2/serverInfo HTTP/1.1" 200 None
DEBUG:requests_kerberos.kerberos_:handle_other(): Handling: 200
DEBUG:requests_kerberos.kerberos_:handle_response(): returning <Response [200]>
Out[1]: 'anonymous'
```

Apparently something is wrong here.
I tried to check the sources of python jira a bit, and I saw  that if I remove the OPTIONAL for the mutual_authentication I get error to authenticate.
https://github.com/pycontribs/jira/blob/develop/jira/client.py#L2284

Am I doing something wrong or is kerberos authentication broken?

Thanks in advance

**Comment by hdost** [comment]
@KKoukiou we recently merged https://github.com/pycontribs/jira/pull/472 into develop can you test with that and see if it fixes your issue?

**Comment by KKoukiou** [comment]
@hdost, still can't authenticate from kerberos. Nothing changed in the logs, from the post above.

Also I took a look in the PR you sent me and looks problematic from a first sight.
https://github.com/pycontribs/jira/blob/69fac2d1325589a37b5ab40542d1b6b2ca5ff418/jira/client.py#L2310-L2314
Is it correct that the only available option would be to disable mutual authentication?
Not setting at all kerberos_options will fail for example, or setting mutual_authentication to OPTIONAL will fail as well.

**Comment by hdost** [comment]
@pbabinca has volunteered to take up this issue

**Comment by pbabinca** [comment]
@KKoukiou can you test if authentication works with referenced PR?

Interestingly in my case when is call `jira.session().__dict__` I can see that I'm actually authenticated.

**Comment by KKoukiou** [comment]
So maybe I am doing something wrong :)
If jira.current_user() does not give `anonymous` for you, feel free to close the issue!

**Comment by pbabinca** [comment]
It does. But if I run:
```
jira.session()
jira.current_user()
```
second call doesn't print `anonymous` anymore. I'm not sure if that's intended for other authentications though.

**Comment by ashcrow** [comment]
I also am seeing the same results as @KKoukiou 

```
$ pip freeze | grep requests
requests==2.18.4
requests-kerberos==0.12.0
requests-oauthlib==0.8.0
requests-toolbelt==0.8.0
$ pip freeze | grep jira
jira==1.0.10
```

When having a valid kerberos ticket and able to use it via the web ui the following happens when calling ```jira.session()```

**Note**: I turned on logging in ``requests_kerberos``.

```
In [8]: jira.session()
DEBUG:requests_kerberos.kerberos_:handle_401(): Handling: 401
DEBUG:requests_kerberos.kerberos_:handle_401(): Kerberos is not supported
DEBUG:requests_kerberos.kerberos_:handle_401(): returning <Response [401]>
....
```

Obviously kerberos is supported since it's being used in the browser, but for some reason it seems ``requests_kerberos`` is believing that the server does not support it. The actual area of ``requests_python`` that is deciding that kerberos is unsupported is [here](https://github.com/requests/requests-kerberos/blob/master/requests_kerberos/kerberos_.py#L285-L292). I believe it is not finding the expected negotiate value in the "www-authenticate" header. Doing a plain ``requests`` call shows this as there is no negotiate in it's "WWW-authenticate" response.

Using ``force_preemptive=True`` in ``requests_kerberos`` does send up the proper negotiate header, but the server rejects it. It's quite odd.

**tl;dr**

I think this is a server misconfiguration of some kind.
